numcodecs/delta.py

```diff
@@ -67,7 +67,7 @@ def encode(self, buf):
         if arr.dtype == bool:
             np.not_equal(arr[1:], arr[:-1], out=enc[1:])
         else:
-            np.subtract(arr[1:], arr[:-1], out=enc[1:])
+            np.subtract(arr[1:], arr[:-1], out=enc[1:], casting='unsafe')
```

Comment:
I added overflow tests for floating points.
I left comments on why the integer overflow tests are not included on the `test_delta.py`.
More specifically, silent integer overflow is the limitation of Numpy (see [numpy/numpy#8987](https://github.com/numpy/numpy/issues/8987)).
I don't have to change Delta's document because the possible silent failure has already been described.

Reply:
Could you elaborate on what need to check about `decode`?